src/devhostagent.restorationjob/RestorationJobApp.cs

```diff
@@ -142,10 +142,17 @@ private async Task<ExitCode> ExecuteInnerAsync(CancellationToken cancellationTok
                             perfLogger.SetProperty(HasConnectedClients, false);
 
                             var disconnectedTimeSpan = DateTimeOffset.Now - lastPingWithSessions;
-                            if (disconnectedTimeSpan != null && disconnectedTimeSpan.Value > _restorationJobEnvironmentVariables.RestoreTimeout)
+                            if ((disconnectedTimeSpan != null && disconnectedTimeSpan.Value > _restorationJobEnvironmentVariables.RestoreTimeout) || lastPingWithSessions == null)

```

Comment:
I think we should add a count check here for safety, just in case when bridge is starting this condition may ever happen first time around. If lastPingWithSessions has been null with no connected sessions 3 times then restore. It solves issue same way just in third iteration of this loop.

Reply:
@elenavillamil Thanks for the suggestion. Instead of fixed count, I thought we will rely on existing restore timeout and calculate the time until lastPingWithSessions is null and if that exceeds restore timeout then it would restore back to original state. Updated UT's accordingly. Please let me know.